### PR TITLE
added pre-commit files

### PR DESCRIPTION
**Pull request by mbeach42**
added pre-commit files

### CODE REVIEW

**Comment by mbeach42** [comment]
@mstechly 
1. it's conventional to have it in the main directory. it's got a ., so its a hidden file anyway.
2. I added info how to install, and added it to setup.py as a test dependancy. So we can close that other PR.

**Review by mstechly — APPROVED** [review]

**Comment by mstechly** [comment]
@mbeach42 Ok, sounds good then!

File: README.md [hunk 1]
```diff
@@ -46,29 +49,42 @@ Once that is done you can:
...
+save_circuit(circuit, "circuit.json")
 ```
 
 Even though it's intended to be used with Orquestra, `z-quantum-core` can be also used as a standalone Python module.
-To install it, you just need to run `pip install -e .` from the main directory.
+To install it, you just need to run `pip install -e .` from the main directory. )
 
 ## Development and Contribution
 
+To install the development version, run `pip install -e .'[test]'` from the main directory. (if using MacOS, you will need single quotes around the []. If using windows, or Linux, you might not need the quotes)
```
Comment (alexjuda) [inline README.md:78]:
```suggestion
To install the development version, run `pip install -e '.[dev]'` from the main directory. (if using MacOS, you will need single quotes around the []. If using windows, or Linux, you might not need the quotes)
```

**Review by alexjuda — APPROVED** [review]
Besides that leftover snippet, LGTM

**Pull request merged**